src/components/App.tsx

```diff
@@ -84,26 +67,73 @@ const App = (): JSX.Element => {
+        {/* pages with personal info */}
         <Route
-          path={BOOKMARKED_ITEMS_PATH}
-          element={<FavoriteWithAuthorization />}
-        />
+          element={
+            // redirect to sign in if not signed in
+            <SignedInWrapper
+              currentAccount={currentAccount}
+              redirectionLink={buildSignInPath({
+                host: GRAASP_AUTH_HOST,
+                redirectionUrl: window.location.toString(),
+              })}
+              onRedirect={() => {
+                // save current url for later redirection after sign in
+                saveUrlForRedirection(pathname, DOMAIN);
+              }}
+            >
+              <PreventGuestWrapper
+                id={PREVENT_GUEST_MESSAGE_ID}
+                currentAccount={currentAccount}
+                buttonText={translateBuilder(BUILDER.GUEST_SIGN_OUT_BUTTON)}
+                onButtonClick={() => signOut()}
+                text={
+                  <Trans
+                    t={translateBuilder}
+                    i18nKey={BUILDER.GUEST_LIMITATION_TEXT}
+                    values={{
+                      name: currentAccount?.name,
+                    }}
+                    components={{ 1: <strong /> }}
+                  />
+                }
+              >
+                <Outlet />
+              </PreventGuestWrapper>
+            </SignedInWrapper>
+          }
+        >
+          <Route path={HOME_PATH} element={<HomeScreen />} />
+          <Route
+            path={BOOKMARKED_ITEMS_PATH}
+            element={<BookmarkedItemsScreen />}
+          />
+          <Route
+            path={PUBLISHED_ITEMS_PATH}
+            element={<PublishedItemsScreen />}
+          />
+          <Route path={RECYCLE_BIN_PATH} element={<RecycledItemsScreen />} />
+        </Route>
+
+        {/* item pages - can be public */}
         <Route
-          path={PUBLISHED_ITEMS_PATH}
-          element={<PublishedWithAuthorization />}
-        />
-        <Route path={buildItemPath()} element={<ItemScreenLayout />}>
+          path={buildItemPath()}
+          element={
+            <ItemLoginWrapper>
+              <ItemScreenLayout />
+            </ItemLoginWrapper>
+          }
+        >
           <Route index element={<ItemScreen />} />
           <Route element={<ItemPageLayout />}>
             <Route path={ITEM_SHARE_PATH} element={<ItemSharingPage />} />
             <Route path={ITEM_PUBLISH_PATH} element={<LibrarySettingsPage />} />
             <Route path={ITEM_SETTINGS_PATH} element={<ItemSettingsPage />} />
           </Route>
         </Route>
-        <Route path={RECYCLE_BIN_PATH} element={<RecycleWithAuthorization />} />
-        <Route path={buildItemPath()} element={<ItemScreen />} />
-        <Route path={ITEMS_PATH} element={<HomeWithAuthorization />} />
+
+        {/* redirection to home */}
+        <Route path={ITEMS_PATH} element={<Redirect />} />
```

Comment:
What is this item path ?

Reply:
I guess it's /items that does not have any id, so you return to home.

Reply:
This will be handled by the "catchall" route defined at the end no ?